src/main/scala/com/cluster/graph/GlobalCoordinator.scala

```diff
@@ -0,0 +1,149 @@
+  var doneCounter = collection.mutable.Map[Int, Int]().withDefaultValue(0)
+  // counts the number of partitions that have voted to terminate their computation
+  var voteCounter = collection.mutable.Map[Int, Int]().withDefaultValue(0)
+  var numPartitions = -1
+  var numNodes = -1
+
+  def globallyDone(stepNum: Int): Boolean = {
+    doneCounter(stepNum) + voteCounter(stepNum) == numPartitions
+  }
+
+  def globallyTerminated(stepNum: Int): Boolean = {
+    voteCounter(stepNum) == numPartitions
+  }
+
+  def broadcastBEGINToPCs(stepNum: Int) = {
+    for ((pid, pcRef) <- pcRefs) {
+      pcRef ! PartitionCoordinator.BEGIN(stepNum)
+    }
+  }
+
+  override def onMessage(msg: GlobalCoordinator.Command): Behavior[GlobalCoordinator.Command] = {
+    msg match {
+
+      case Initialize(pcs, ns, replyTo) =>
+        pcRefs ++= pcs
+        numNodes = ns
+        numPartitions = pcRefs.size
+        replyTo ! InitResponse("Initialized the Global Coordinator")
+        Behaviors.same
+
+      case BEGIN() =>
+        broadcastBEGINToPCs(0)
+        Behaviors.same
+
+      case DONE(stepNum) =>
+        doneCounter(stepNum) += 1
+        if (globallyDone(stepNum)) {
+          broadcastBEGINToPCs(stepNum + 1)
+        }
+        Behaviors.same
+
+      case TerminationVote(stepNum) =>
+        voteCounter(stepNum) += 1
+        if (globallyTerminated(stepNum)) {
+          //TODO TERMINATE..?
+        } else if (globallyDone(stepNum)) {
+          broadcastBEGINToPCs(stepNum + 1)
+        }
+        Behaviors.same
+
+      case BroadcastRef(gcRef, replyTo) =>
+        for ((pid, pcRef) <- pcRefs) {
+          val f: Future[PartitionCoordinator.UpdateGCResponse] = pcRef.ask(ref => {
+            PartitionCoordinator.UpdateGC(gcRef, ref)
+          })
+          val UpdateGCResponse = Await.result(f, waitTime)
+          UpdateGCResponse match {
+            case PartitionCoordinator.UpdateGCResponse(message) =>
+              println(message)
+            case _ =>
+              println(s"PC${pid} did not receive GCRef")
+              replyTo ! BroadcastRefResponse("GC Broadcast did not complete")
+              Behaviors.same
+          }
+        }
+        replyTo ! BroadcastRefResponse("GC Broadcast to all PCs!")
+        Behaviors.same
+
+      case GetPCRefs(replyTo) =>
+        replyTo ! GetPCRefsResponse(pcRefs)
+        Behaviors.same
+
+    }
+  }
+}
+
+object GlobalCoordinator {
+
+  val GlobalCoordinatorKey = ServiceKey[GlobalCoordinator.Command](s"globalCoordinator")
+
+  def apply(): Behavior[GlobalCoordinator.Command] = {
+    Behaviors.setup(ctx => {
+
+      ctx.system.receptionist ! Receptionist.Register(GlobalCoordinatorKey, ctx.self)
+      new GlobalCoordinator(ctx)
+    })
+  }
+
+  sealed trait Response extends CborSerializable
+
+  trait Command extends CborSerializable
+
+  case class DONE(stepNum: Int) extends Command
+  case class TerminationVote(stepNum: Int) extends Command
+  case class BEGIN() extends Command
```

Comment:
Nit: Could just be final case object?

Reply:
Agreed. This seems important:
- https://doc.akka.io/docs/akka/current/typed/persistence-style.html
- https://nrinaudo.github.io/scala-best-practices/tricky_behaviours/final_case_classes.html

I'll modify all `Command`, `Response` traits to `sealed`, modify all such extended case classes to `final`, and merge.